Animated filter

carmex opened the pull request:
feat: Filter gisa results for GIFs and improve documentation

This commit introduces a key improvement to the 'gisa' (animated gif) search functionality. Previously, a 'gisa' search could return non-GIF image types. Now, the results are explicitly filtered to ensure only images with a .gif extension are returned.

Key changes:
- The command parser in `src/parser/index.ts` has been updated to include the search modifier (e.g., 'a' for 'gisa') in its output.
- The server logic in `src/server/service.ts` now uses this modifier to apply a .gif file extension filter on the results from Google Image Search for all 'gisa' commands.
- Added a new test case to `src/server/service.test.ts` to verify the GIF filtering behavior.
- Updated existing parser tests to align with the new return structure.

Additionally, the project's README.md has been significantly expanded with detailed sections on:
- Features and command modifiers
- Configuration and environment variables
- Instructions for building, running (locally and with Docker), and testing
- An overview of the codebase for new developers.

carmex committed:
feat: Filter gisa results for GIFs and improve documentation

This commit introduces a key improvement to the 'gisa' (animated gif) search functionality. Previously, a 'gisa' search could return non-GIF image types. Now, the results are explicitly filtered to ensure only images with a .gif extension are returned.

Key changes:
- The command parser in `src/parser/index.ts` has been updated to include the search modifier (e.g., 'a' for 'gisa') in its output.
- The server logic in `src/server/service.ts` now uses this modifier to apply a .gif file extension filter on the results from Google Image Search for all 'gisa' commands.
- Added a new test case to `src/server/service.test.ts` to verify the GIF filtering behavior.
- Updated existing parser tests to align with the new return structure.

Additionally, the project's README.md has been significantly expanded with detailed sections on:
- Features and command modifiers
- Configuration and environment variables
- Instructions for building, running (locally and with Docker), and testing
- An overview of the codebase for new developers.

carmex committed:
updated pre-commit hook to work better

codingismy11to7 commented on src/server/service.ts:
```diff
@@ -15,11 +15,13 @@ class InvalidSearch {
   /(alamy\.com)|(depositphotos\.com)|(shutterstock\.com)|(maps\.google\.com)|(fbsbx.*\.com)|(memegenerator.*\.net)|(gstatic.*\.com)|(instagram.*\.com)|(tiktok.*\.com)/i;
 
-const doSearch = (text: string, index: number, giser: GISerFuncs) =>
+const doSearch = (text: string, index: number, mod: "g" | "t" | "i" | "a" | "m" | "l" | undefined, giser: GISerFuncs) =>
```
this needs to use the `Mod` type, which will need exporting

carmex committed:
refactor: Use shared Mod type for search parameters

Exports the 'Mod' type from the parser and uses it in the server's 'doSearch' function. This addresses a PR review comment to improve type safety by removing a duplicated inline type definition.
